File: app/src/main/java/me/wcy/music/utils/binding/ViewBinder.java
package me.wcy.music.utils.binding;

import android.app.Activity;
import android.view.View;

import java.lang.reflect.Field;

public class ViewBinder {

    public static void bind(Activity activity) {
        bind(activity, activity.getWindow().getDecorView());
    }

    public static void bind(Object target, View source) {
        Field[] fields = target.getClass().getDeclaredFields();
        if (fields != null && fields.length > 0) {
            for (Field field : fields) {
                try {
                    field.setAccessible(true);
                    if (field.get(target) != null) {
                        continue;
                    }

                    Bind bind = field.getAnnotation(Bind.class);
                    if (bind != null) {
                        int viewId = bind.value();
                        field.set(target, source.findViewById(viewId));
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: app/src/main/java/me/wcy/music/utils/binding/ViewBinder.kt
package me.wcy.music.utils.binding

import android.app.Activity
import android.view.View

object ViewBinder {
    fun bind(activity: Activity) {
        bind(activity, activity.window.decorView)
    }

    fun bind(target: Any, source: View?) {
        val fields = target.javaClass.declaredFields
        if (fields.isNotEmpty()) {
            for (field in fields) {
                try {
                    field.isAccessible = true
                    if (field[target] != null) {
                        continue
                    }
                    val bind = field.getAnnotation(
                        Bind::class.java
                    )
                    if (bind != null) {
                        val viewId = bind.value
                        field[target] = source!!.findViewById(viewId)
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        }
    }
}